pkg/fleet-manager/backup_controller.go

```diff
@@ -0,0 +1,291 @@
+		WithOptions(options).
+		Complete(b)
+}
+
+func (b *BackupManager) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
+	log := ctrl.LoggerFrom(ctx)
+	backup := &backupapi.Backup{}
+
+	if err := b.Client.Get(ctx, req.NamespacedName, backup); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("backup object not found", "backup", req)
+			return ctrl.Result{}, nil
+		}
+
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	// Initialize patch helper
+	patchHelper, err := patch.NewHelper(backup, b.Client)
+	if err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "failed to init patch helper for backup %s", req.NamespacedName)
+	}
+	// Setup deferred function to handle patching the object at the end of the reconciler
+	defer func() {
+		patchOpts := []patch.Option{}
+		if err := patchHelper.Patch(ctx, backup, patchOpts...); err != nil {
+			reterr = utilerrors.NewAggregate([]error{reterr, errors.Wrapf(err, "failed to patch %s  %s", backup.Name, req.NamespacedName)})
+		}
+	}()
+
+	// Check and add finalizer if not present
+	if !controllerutil.ContainsFinalizer(backup, BackupFinalizer) {
+		controllerutil.AddFinalizer(backup, BackupFinalizer)
+		return ctrl.Result{}, nil
+	}
+
+	// Handle deletion
+	if backup.GetDeletionTimestamp() != nil {
+		return b.reconcileDeleteBackup(ctx, backup)
+	}
+
+	// Handle the main reconcile logic
+	return b.reconcileBackup(ctx, backup)
+}
+
+// reconcileBackup handles the main reconcile logic for a Backup object.
+func (b *BackupManager) reconcileBackup(ctx context.Context, backup *backupapi.Backup) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// Fetch destination clusters
+	destinationClusters, err := fetchDestinationClusters(ctx, b.Client, backup.Namespace, backup.Spec.Destination)
+	if err != nil {
+		log.Error(err, "failed to fetch destination clusters for backup", "backupName", backup.Name)
+		return ctrl.Result{}, err
+	}
+	// Apply velero backup resource in target clusters
+	result, err := b.reconcileBackupResources(ctx, backup, destinationClusters)
+	if err != nil || result.Requeue || result.RequeueAfter > 0 {
+		return result, err
+	}
+	// Collect velero backup resource status to current backup
+	return b.reconcileBackupStatus(ctx, backup, destinationClusters)
+}
+
+// reconcileBackupResources converts the backup resources into velero backup resources that can be used by Velero on the target clusters, and applies each of these backup resources to the respective target clusters.
+func (b *BackupManager) reconcileBackupResources(ctx context.Context, backup *backupapi.Backup, destinationClusters map[ClusterKey]*fleetCluster) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	backupLabel := generateVeleroInstanceLabel(BackupNameLabel, backup.Name, backup.Spec.Destination.Fleet)
+
+	if isScheduleBackup(backup) {
+		// Handle scheduled backups
+		for clusterKey, clusterAccess := range destinationClusters {
```

Comment:
For perf: We can sync backup resource to clusters concurrently

Reply:
ok